src/pymmcore_widgets/_shutter_widget.py

```diff
@@ -0,0 +1,232 @@
+from __future__ import annotations
+
+from typing import Any, Optional, Tuple, Union
+
+from fonticon_mdi6 import MDI6
+from pymmcore_plus import CMMCorePlus, DeviceType
+from qtpy import QtWidgets as QtW
+from qtpy.QtCore import QSize, Qt
+from qtpy.QtGui import QColor, QIcon
+from superqt.fonticon import icon
+from superqt.utils import signals_blocked
+
+from ._core import get_core_singleton
+
+COLOR_TYPE = Union[
+    QColor,
+    int,
+    str,
+    Qt.GlobalColor,
+    Tuple[int, int, int, int],
+    Tuple[int, int, int],
+]
+
+
+class ShuttersWidget(QtW.QWidget):
+    """A Widget for shutters and Micro-Manager autoshutter.
+
+    Parameters
+    ----------
+    shutter_device: str:
+        The shutter device Label.
+    autoshutter: bool
+        If True, a checkbox controlling the Micro-Manager autoshutter
+        is added to the layout.
+    button_text_open_closed: Optional[tuple[str, str]]
+       Text of the QPushButton when the shutter is open or closed
+    icon_size : Optional[str]
+        Size of the QPushButton icon.
+    icon_color_open_closed : Optional[tuple[COLOR_TYPE, COLOR_TYPE]]
+        Color of the QPushButton icon when the shutter is open or closed.
+    text_color_combo:
+        Text color of the shutter QComboBox.
+    parent : Optional[QWidget]
+        Optional parent widget.
+
+    COLOR_TYPE = Union[QColor, int, str, Qt.GlobalColor, Tuple[int, int, int, int],
+    Tuple[int, int, int]]
+    """
+
+    def __init__(
+        self,
+        shutter_device: str,
+        autoshutter: bool = True,
+        button_text_open_closed: Tuple[str, str] = (None, None),  # type: ignore
+        icon_open_closed: Tuple[QIcon, QIcon] = (
+            MDI6.hexagon_outline,
+            MDI6.hexagon_slice_6,
+        ),
+        icon_size: int = 25,
+        icon_color_open_closed: tuple[COLOR_TYPE, COLOR_TYPE] = ("", ""),
+        text_color_combo: COLOR_TYPE = "",
```

Comment:
i dislike having all of these style-related things in the `__init__` method.  Also, it looks like at least `text_color_combo` is never used anywhere.  Can you

1) remove these from the init, and expose setter methods for the properties that you feel _must_ be exposed (but don't recreate any setters than Qt already provides... we don't want to create any additional APIs that aren't strictly necessary)
2) review them all to make sure they are used?

Reply:
side note: this won't be necessary after you remove all the params from the `__init__`, but if you have this many parameters in a function call, you should generally try to make them all keyword only (i.e. move the `*` farther up in the call).  This allows you to reposition them later without breaking anyone's usage (if they happened to be calling them positionally)